pkg/cmd/readyz/readyz.go

```diff
@@ -0,0 +1,205 @@
+package readyz
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"syscall"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"go.etcd.io/etcd/client/pkg/v3/transport"
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"golang.org/x/sys/unix"
+	"google.golang.org/grpc"
+	"k8s.io/apiserver/pkg/server"
+	"k8s.io/klog/v2"
+)
+
+const (
+	defaultListenPort      = 9980
+	defaultHTTPDialTimeout = 2 * time.Second
+	defaultEndpoint        = "https://localhost:2379"
+	// keepalive defaults used by kube-apiserver
+	// https://github.com/kubernetes/apiserver/blob/de6ba2aa0a752d077719fcee186291d7af73a825/pkg/storage/storagebackend/factory/etcd3.go#L55-L56
+	keepaliveTime    = 30 * time.Second
+	keepaliveTimeout = 10 * time.Second
+)
+
+type readyzOpts struct {
+	listenPort       uint16
+	dialTimeout      time.Duration
+	targetEndpoint   string
+	servingCertFile  string
+	servingKeyFile   string
+	clientCertFile   string
+	clientKeyFile    string
+	clientCACertFile string
+}
+
+func newReadyzOpts() *readyzOpts {
+	return &readyzOpts{
+		listenPort:     defaultListenPort,
+		dialTimeout:    defaultHTTPDialTimeout,
+		targetEndpoint: defaultEndpoint,
+	}
+}
+
+// NewReadyzCommand creates a readyz command that runs as an http-get readiness server alongside the etcd member container
+func NewReadyzCommand() *cobra.Command {
+	opts := newReadyzOpts()
+	cmd := &cobra.Command{
+		Use:   "readyz",
+		Short: "Serve the HTTP /readyz endpoint health check for an etcd member",
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := opts.Validate(); err != nil {
+				klog.Fatal(err)
+			}
+			if err := opts.Run(); err != nil {
+				klog.Fatal(err)
+			}
+		},
+	}
+
+	opts.AddFlags(cmd.Flags())
+	return cmd
+}
+
+func (r *readyzOpts) AddFlags(fs *pflag.FlagSet) {
+	fs.Uint16Var(&r.listenPort, "listen-port", r.listenPort, "Listen on this port. Default 9980")
+	fs.DurationVar(&r.dialTimeout, "dial-timeout", defaultHTTPDialTimeout, "Dial timeout for the client. Default 2s")
```

Comment:
default should be r.dialTimeout and use `newReadyzOpts` to set it to what you want